app.js

```diff
@@ -38,9 +38,103 @@ function operate(operator, a, b) {
 const display = document.querySelector(".display");
 const btns = document.querySelectorAll("button");
 
-// put number selected on the display
 btns.forEach((button) => {
-  button.addEventListener("click", (e) => {
-    display.innerHTML += button.innerText;
-  });
+  button.addEventListener("click", (e) => calculatorDisplay(button.id));
 });
+
+let firstNum = "";
+let secondNum = "";
+let operator;
+// put number selected on the display
+function calculatorDisplay(value) {
+  function getData(value) {
+    if (
+      display.innerHTML.includes("-") ||
+      display.innerHTML.includes("+") ||
+      display.innerHTML.includes("x") ||
+      display.innerHTML.includes("÷")
+    ) {
+      secondNum += value;
+    } else {
+      firstNum += value;
+    }
+  }
+
+  switch (value) {
+    case "zero":
+      display.innerHTML += "0";
+      getData("0");
+      break;
+    case "one":
+      display.innerHTML += "1";
+      getData("1");
+      break;
+    case "two":
+      display.innerHTML += "2";
+      getData("2");
+      break;
+    case "three":
+      display.innerHTML += "3";
+      getData("3");
+      break;
+    case "four":
+      display.innerHTML += "4";
+      getData("4");
+      break;
+    case "five":
+      display.innerHTML += "5";
+      getData("5");
+      break;
+    case "six":
+      display.innerHTML += "6";
+      getData("6");
+      break;
+    case "seven":
+      display.innerHTML += "7";
+      getData("7");
+      break;
+    case "eight":
+      display.innerHTML += "8";
+      getData("8");
+      break;
+    case "nine":
+      display.innerHTML += "9";
+      getData("9");
+      break;
+    case "decimal":
+      display.innerHTML += ".";
+      getData(".");
+      break;
+    case "clear":
+      display.innerHTML = "";
+      firstNum = "";
+      secondNum = "";
+      operator = undefined;
+      break;
+    case "minus":
+      display.innerHTML += "-";
+      operator = "subtract";
+      break;
+    case "plus":
+      display.innerHTML += "+";
+      operator = "add";
+      break;
+    case "times":
+      display.innerHTML += "x";
+      operator = "multiply";
+      break;
+    case "divide":
+      display.innerHTML += "÷";
+      operator = "divide";
+      break;
+    case "equals":
+      console.log(firstNum);
+      console.log(secondNum);
+      display.innerHTML = operate(
+        operator,
+        Number(firstNum),
+        Number(secondNum)
+      );
+      break;
```

Comment:
Here you're doing something weird: you assign a word to operator, but later you call operate that will just switch case over these words to choose a function. In javascript, functions are first-class: you can assign them directly to variables.

So instead, you can do something like
```js
operator = divide; // This is the divide *function*, not a string
...
case "equals":
      if (!operator) {
          console.warn("no operator selected");
          return;
      }
      display.innerHTML = operator(Number(firstNum), Number(secondNum));
```

And again, all your operator cases do basically the same thing, so it's probably better to use an if statement:

```js
// Object associating symbols with operations
const operators = {"-": subtract, "+": add, "x" multiply, "÷": divide};
if (Object.keys(operators).includes(value)) {
  display.innerHTML += value;
  operator = operators[value];
}
```

At this point you basically have only 2 cases in your switch: equals and clear, so it might not be worth to keep it as a switch. These changes replace 92 lines with 19. It basically halves the size of your code.